docs/0003-hashed-n-tuple-trees.md

```diff
@@ -0,0 +1,260 @@
+
+Using this extension, OCFL object identifiers are hashed and encoded as hex strings. These digests are then divided into _N_ n-tuple segments, which are used to create nested paths under the OCFL storage root.
+
+This approach allows OCFL object identifiers of any composition to be evenly distributed across the storage hierarchy. The maximum number of files under any given directory is controlled by the number of characters in each n-tuple, and the tree depth is controlled by the number of n-tuple segments each digest is divided into. Additionally, it obviates the need to handle special characters in OCFL object identifiers because the mapped directory names will only ever contain the characters `0-9a-f`.
+
+However, this comes at the cost of not being able to identify the OCFL object identifier of an object simply by browsing the OCFL storage hierarchy. The ID of an object may only be found within its `inventory.json`.
+
+## Parameters
+
+### Summary
+
+* **Name:** `digestAlgorithm`
+  * **Description:** The digest algorithm to apply on the OCFL object identifier; MUST be an algorithm that is allowed in the OCFL fixity block
+  * **Type:** string
+  * **Range:** 1,4095
+  * **Default:** sha256
+* **Name:** `caseMapping`
+  * **Description:** Indicates the casing to use for the hex encoded digest
+  * **Type:** enumerated
+  * **Range:** "toUpper","toLower"
+  * **Default:** "toLower"
+* **Name**: `tupleSize`
+  * **Description:** Indicates the size of the segments (in characters) that the digest is split into
+  * **Type:** integer
+  * **Range:** 0,32
+  * **Default:** 3
+* **Name:** `numberOfTuples`
+  * **Description:** Indicates how many segments are used for path generation
+  * **Type:** integer
+  * **Range:** 0,32
+  * **Default:** 3
+* **Name:** `shortObjectRoot`
+  * **Description:** When true, indicates that the OCFL object root directory name should contain the remainder of the digest not used in the n-tuples segments
+  * **Type:** boolean
+  * **Default:** false
+
+### Details
+
+#### digestAlgorithm
+
+`digestAlgorithm` is defaulted to `sha256`,and it MUST either contain a digest algorithm that's [officially supported by the OCFL spec](https://ocfl.io/draft/spec/#digest-algorithms) or defined in a community extension. The specified algorithm is applied to OCFL object identifiers to produce hex encoded digest values that are then mapped to OCFL object root paths.
+
+#### caseMapping
+
+Hex numbers are case insensitive, however many filesystems are not. To avoid case related problems, all hex encoded digest values MUST be cased consistently. When `caseMapping` is set to `toLower`, the default, all letters in the digest MUST be converted to lowercase. When it's set to `toUpper`, all letters MUST be converted to uppercase.
```

Comment:
Are users wanting to be able to specify this? Can we get away with just forcing it to lower case, and see if anyone needs something different?

Reply:
> Are users wanting to be able to specify this? Can we get away with just forcing it to lower case, and see if anyone needs something different?

That's certainly a possibility. I don't have a strong opinion either way. The core spec doesn't take a stance hex casing other than to say that it should be consistent.

Reply:
@bcail I removed the `caseMapping` parameter.